src/sagemaker_xgboost_container/algorithm_mode/train.py

```diff
@@ -296,19 +308,28 @@ def get_callbacks_watchlist(train_cfg, train_dmatrix, val_dmatrix, model_dir, ch
         logging.info("Resuming from iteration %s", iteration)
 
     callbacks = []
-    callbacks.append(checkpointing.print_checkpointed_evaluation(start_iteration=iteration))
```

Comment:
Should we remove the `print_checkpointed_evaluation` function and its related functions/classes if they're no longer being used? Same thing for `save_checkpoint` and `save_intermediate_model`.

Reply:
They could be used in script mode. Note that they are in the top level module, not just in algorithm mode. So it is better to keep for now.